tidy3d/components/simulation.py

```diff
@@ -2984,3 +2986,198 @@ def from_scene(cls, scene: Scene, **kwargs) -> Simulation:
+            new_bounds = [
+                [
+                    grids_1d.x[grid_inds[0][0]],
+                    grids_1d.y[grid_inds[1][0]],
+                    grids_1d.z[grid_inds[2][0]],
+                ],
+                [
+                    grids_1d.x[grid_inds[0][1]],
+                    grids_1d.y[grid_inds[1][1]],
+                    grids_1d.z[grid_inds[2][1]],
+                ],
+            ]
+
+        # if symmetry is not overriden we inherit it from the original simulation where is needed
+        if symmetry is None:
+
+            # start with no symmetry
+            symmetry = [0, 0, 0]
+
+            # now check in each dimension whether we cross symmetry plane
+            for dim in range(3):
+                if self.symmetry[dim] != 0:
+                    crosses_symmetry = (
+                        new_bounds[0][dim] < self.center[dim]
+                        and new_bounds[1][dim] > self.center[dim]
+                    )
+
+                    # inherit symmetry only if we cross symmetry plane, otherwise we don't impose
+                    # symmetry even if the original simulation had symmetry
+                    if crosses_symmetry:
+                        symmetry[dim] = self.symmetry[dim]
+                        center = (new_bounds[0][dim] + new_bounds[1][dim]) / 2
+
+                        if not math.isclose(center, self.center[dim]):
+                            log.warning(
+                                f"The original simulation is symmetric along {'xyz'[dim]} direction. "
+                                "The requested new simulation region does cross the symmetry plane but is "
+                                "not symmetric with respect to it. To preserve correct symmetry, "
+                                "the requested simulation region is expanded symmetrically."
+                            )
+                            new_bounds[0][dim] = 2 * self.center[dim] - new_bounds[1][dim]
+
+        # symmetry and grid spec treatments could change new simulation bounds
+        # thus, recreate a box instance
+        new_box = Box.from_bounds(*new_bounds)
+
+        # inheritance of structures, sources, monitors, and boundary specs
+        if remove_outside_structures:
+            new_structures = [strc for strc in self.structures if new_box.intersects(strc.geometry)]
+        else:
+            new_structures = self.structures
+
+        if sources is None:
+            sources = [src for src in self.sources if new_box.intersects(src)]
+
+        if monitors is None:
+            monitors = [mnt for mnt in self.monitors if new_box.intersects(mnt)]
+
+        if boundary_spec is None:
+            boundary_spec = self.boundary_spec
+
+        # reduction of custom medium data
+        new_sim_medium = self.medium
+        if remove_outside_custom_mediums:
+
+            # check for special treatment in case of PML
+            if isinstance(boundary_spec, (PML, StablePML, Absorber)):
```

Comment:
I think these three are all instances of `AbsorberSpec`, which itself is a `BoundaryEdge`, not a `BoundarySpec`. something seems a little strange to me, would a `simulation.boundary_spec` ever be an instance of `PML`, etc?

Reply:
oh, good catch, this does seem wrong and I didn't have a test for that. Will fix it, thanks!